src/shims/io.rs

```diff
@@ -42,16 +42,38 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
 
         let flag = this.read_scalar(flag_op)?.to_i32()?;
 
-        if flag != this.eval_libc_i32("O_RDONLY")? && flag != this.eval_libc_i32("O_CLOEXEC")? {
-            throw_unsup_format!("Unsupported flag {:#x}", flag);
+        let mut options = OpenOptions::new();
+
+        // The first two bits of the flag correspond to the access mode of the file in linux.
+        let access_mode = flag & 0b11;
```

Comment:
Is this correct for every unix? Maybe it should and O_RDWR instead?

Reply:
I'm not sure about it. But on linux [`O_RDWR` is 2](https://docs.rs/libc/0.2.63/src/libc/unix/linux_like/mod.rs.html#494). So that won't work

Reply:
Didnt know that. Using the same way as O_APPEND and friends should work though.

Reply:
I thought the same, but the thing is that `O_RDONLY` is 0... :/

Reply:
:( At least linux, macos and windows use 0, 1 and 2, so I guess we can keep it this way.

Reply:
Yeah... And 3 is basically here-be-dragons... everything should be shifted by 1 D: